src/plugins/intel_cpu/src/transformations/snippets/x64/pass/lowered/adjust_brgemm_copy_b_loop_ports.cpp

```diff
@@ -65,17 +65,27 @@ bool pass::AdjustBrgemmCopyBLoopPorts::run(const snippets::lowered::LinearIR& li
 
     bool modified = false;
 
+    auto get_repacking_loop_idces = [](const snippets::lowered::ExpressionPtr& parent_expr) {
+        // Repacking may be extracted outside the snippets kernel. In this case, brgemm parent expression is a parameter.
+        if (is_type<ov::op::v0::Parameter>(parent_expr->get_node()))
+            return std::vector<size_t>{};
+
+        OPENVINO_ASSERT(is_type<snippets::lowered::BufferExpression>(parent_expr),
+                        "In case of repacking brgemm expr must have BufferExpression on B input");
+        const auto buffer_parent_ports = parent_expr->get_input_port(0).get_connected_ports();
+        OPENVINO_ASSERT(buffer_parent_ports.size() == 1,
+                        "Parent of brgemm repacking buffer must be connected only to the buffer");
+        const auto& repacking_expr = buffer_parent_ports.begin()->get_expr();
+        return repacking_expr->get_loop_ids();
+    };
+
     for (const auto& expr : linear_ir) {
-        const auto& node = expr->get_node();
-        if (!is_type<BrgemmCopyB>(node))
+        const auto brgemm = ov::as_type_ptr<BrgemmCPU>(expr->get_node());
+        if (!brgemm || !brgemm_utils::with_repacking(brgemm->get_type()))
             continue;
-        const auto& repacking_loop_ids = expr->get_loop_ids();
-        const auto& child_ports = expr->get_output_port(0).get_connected_ports();
-        OPENVINO_ASSERT(child_ports.size() == 1 &&
-                        is_type<snippets::lowered::BufferExpression>(child_ports.begin()->get_expr()),
-                        "BrgemmCopyB should have one BufferExpression child");
-        auto grandchild_ports = child_ports.begin()->get_expr()->get_output_port(0).get_connected_ports();
-        for (const auto& target_port : grandchild_ports) {
+        const auto& parent_expr = expr->get_input_port_connector(1)->get_source().get_expr();
+        const auto& repacking_loop_ids = get_repacking_loop_idces(parent_expr);
+        for (const auto& target_port : parent_expr->get_output_port(0).get_connected_ports()) {
```

Comment:
Looks like something goes wrong here...

Firstly, we try to find `Brgemm`.  And then we get loop ids of repacking.
Secondly, we iterate through consumers of second input of Brgemm: through the same Brgemm and Loops..
Previously, if I got it correcntly, we iterated through consumers of this Brgemm...

Could you please verify this moment?

I think that we can just iterate through `expr->get_output_connector(0)->get_consumers()` or `expr->get_output_port(0).get_connected_ports()`

Reply:
Thanks for the pointing this out! I refactored the code to avoid useless ports traversal, please take a look